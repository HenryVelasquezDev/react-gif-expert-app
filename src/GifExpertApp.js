import React, {useState} from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
//import PropTypes from 'prop-types'

const GifExpertApp = ({defaultCategories = []}) => {

    //const [categories, setcategories] = useState(['Dragon Ball Z']);
    const [categories, setcategories] = useState(defaultCategories);

    return (
        <>
            <h2>CounterApp</h2>
            <AddCategory SetCategories = { setcategories }/>
            <hr/>
            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key= { category }
                            category= { category }
                        />
                    ))
                }
            </ol>
        </>
    );
}

/*GifExpertApp.propTypes = {

}*/

export default GifExpertApp;