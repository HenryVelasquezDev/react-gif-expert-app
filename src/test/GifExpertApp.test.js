import React  from 'react';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en <GifExpertApp /> ', () => {
    
    test('debe de mostrarse correctamente ', () => {
        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias ', () => {
        const categories = ['Dragon Ball Z','Hunter x Hunter'];
        const wrapper = shallow( <GifExpertApp defaultCategories = {categories}/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe( categories.length );

    });
        
    
});
