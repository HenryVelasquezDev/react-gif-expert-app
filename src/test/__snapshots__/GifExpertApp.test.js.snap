// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp />  debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    CounterApp
  </h2>
  <AddCategory
    SetCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
    <GifGrid
      category="Hunter x Hunter"
      key="Hunter x Hunter"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp />  debe de mostrarse correctamente  1`] = `
<Fragment>
  <h2>
    CounterApp
  </h2>
  <AddCategory
    SetCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
